### PR TITLE
Incorporate classnames package and do some general cleanup

### PR DESCRIPTION
**Pull request by jakewheeler**
## Summary

This PR does the following:

- Adds and makes use of `classNames` where appropriate
- Adds some reasonable defaults for `prettier` and `eslint`
  - A couple of new `devDependencies` were added for this 
- Formats all applicable files using the new `prettier` defaults
- Ensures the `Footer` component renders a `footer` element rather than another `header`
- Moves `NavigationLink` and its prop type into its own file
- Moved the heavily used `basePath` variable into its own `constants.ts` file
- Fixed an error where Next.js couldn't find the `favicon.ico`
- Some other minor cleanup 

## Testing

1. Run `npm i` to install the new packages
2. Start the project (restart might be needed)
3. Make sure everything generally still works and looks the same

If you happen to run into any issues, it might be worth trying to delete the `.next` folder and restarting your dev build.

## Discussion

The "reasonable defaults" felt like a good starting point to me but we can definitely adjust these over time if we think any of the ESLint rules aren't useful to us.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,14 +8,15 @@
     "start": "next start",
     "lint": "next lint",
     "prepare": "husky",
-    "prettier": "prettier --write ."
+    "fmt": "prettier --write ."
```
Comment (jakewheeler) [inline package.json:11]: Any issues with this change? It felt easier to type out 😄

File: src/app/components/NavigationLink.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+}: NavigationLinkProps) {
+  const pathname = usePathname(); // Use usePathname hook to get the current path
+  const isActive = pathname === href;
+  return (
+    <Link href={href} passHref className="usa-nav__link" onClick={onClick}>
+      <span
+        className={classNames(styles.navbarItemText, {
+          'navbar-item-active': isActive,
+        })}
+      >
```
Comment (jakewheeler) [inline src/app/components/NavigationLink.tsx:23]: I think this is a good example of where `classNames` is really helpful.

File: package.json (hunk 1 above)
Comment (nickbristow) [reply package.json:11]: 👍

File: src/app/components/NavigationLink.tsx (hunk 2 above)
Comment (nickbristow) [reply src/app/components/NavigationLink.tsx:23]: 👍

**Review by nickbristow — APPROVED** [review]

**Pull request merged**